src/containers/denuncias/ValidaDenuncia.js

```diff
@@ -0,0 +1,140 @@
+}
+
+function temCaractereEspecial(campo) {
+  const format = /[!@#$%^&*()_+-=[\]{};':"\\|,.<>/?]+/;
+  return format.test(campo);
+}
+
+function temTamanhoDeCaracteresMaiorQue(nomeDoCampo, campo, tamanho) {
+  if (campo.length > tamanho) {
+    return `A descrição do campo ${nomeDoCampo} está muito grande. Deve ter menos de ${tamanho} caracteres.`;
+  }
+  return undefined;
+}
+
+function validaCamposVaziosOrNulos(campos) {
+  if (campos === null || campos === '') {
+    return 'Por favor, preencha a denúncia.';
+  }
+  return undefined;
+}
+
+function caracteristicasVitimaEhVazia(caracteristicasDaVitima) {
+  if (caracteristicasDaVitima === '' || caracteristicasDaVitima === null || caracteristicasDaVitima === undefined) {
+    focoNoCampo();
+    return 'Por favor, descreva as características da vítima.';
+  }
+  return undefined;
+}
+
+function validaCaracteriscaVitima(caracteristicasDaVitima) {
+  return (caracteristicasVitimaEhVazia(caracteristicasDaVitima) ||
+         temTamanhoDeCaracteresMaiorQue('caracteristicas da vitima', caracteristicasDaVitima, 255));
+}
+
+function temNumeroNoNome(nome) {
+  if (temNumero(nome)) {
+    return 'Por favor, preencha o campo nome sem números.'
+  }
+  return undefined;
+}
+
+function temCaractereEspecialNoNome(nome) {
+  if (temCaractereEspecial(nome)) {
+    return 'Por favor, preencha o campo nome sem caractere especial.'
+  }
+  return undefined;
+}
+
+function validaNome(nome) {
+  return temNumeroNoNome(nome) ||
+         temCaractereEspecialNoNome(nome) ||
+         temTamanhoDeCaracteresMaiorQue('Nome', nome, 40);
+}
+
+function validaGenero(genero) {
+  return temTamanhoDeCaracteresMaiorQue('Gênero', genero, 15);
+}
+
+function ehDataAtual(data) {
+  const dataHojeFormatada = new Date().toJSON().slice(0, 10);
+  return data >= dataHojeFormatada;
+}
+
+function ehDataAntes1900(data) {
+  const primeiraDataValida = new Date('01/01/1900').toJSON().slice(0, 10);
+  return data < primeiraDataValida;
+}
+
+function validaDataDeNascimento(dataNascimento) {
+  if (dataNascimento.trim() !== '') {
+    const dataNascimentoFormatada = new Date(dataNascimento).toJSON().slice(0, 10);
+    if (ehDataAtual(dataNascimentoFormatada) || ehDataAntes1900(dataNascimentoFormatada)) {
+      return 'Por favor, preencha o campo data de nascimento com uma data válida.'
+    }
+  }
+  return undefined;
+}
+
+function validaEndereco(endereco) {
+  return temTamanhoDeCaracteresMaiorQue('Endereço', endereco, 255);
+}
+
+function ehTelefoneValido(telefone) {
+  return (new RegExp('^[0-9]{10,11}$').test(telefone));
+}
+
+function validaTelefone(telefone) {
+  if (telefone !== '' && !ehTelefoneValido(telefone)) {
+    return 'Por favor, preencha o campo telefone com um número válido.';
+  }
+  return undefined;
+}
+
+function ehEmailValido(valor) {
+  return (/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(valor));
+}
+
+function validaEmail(email) {
+  if (email !== '' && !ehEmailValido(email)) {
```

Comment:
A função ehEmailValido poderia validar se o email informado está vazio ou não, com isso ele ganha mais propósito e toda a validação fica centralizada. Testando o email com email !== '', se email for undefined, essa condição será true.
Ex: return !email && (/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(valor));